Handbook: update release docs (#23002)
WordPress/gutenberg: docs/contributors/release.md

### Schedule

> We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when _the release candidate_ is to be _tagged).

Reviewer: > (the day when _the release candidate_ is to be _tagged).
(the day when _the release candidate_ is to be _tagged_).

Reviewer 2: ```suggestion
We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when _the release candidate_ is to be tagged).
```

> We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (note this is the day of _tagging the release candidate_).

Reviewer 3: ```suggestion
We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date (the day when the release candidate is to be _tagged_).
```

- **On the date of the current milestone**, we publish a release candidate and make it available for plugin authors and users to test. If any regressions are found with a release candidate, a new one can be published. On this date, all remaining PRs on the milestone are moved automatically to the next release. Release candidates should be versioned incrementally, starting with `-rc.1`, then `-rc.2`, and so on.

- **Two days after the first release candidate**, the stable version is created based on the last release candidate and any necessary regression fixes. Once the stable version is released, a post [like this](https://make.wordpress.org/core/2019/06/26/whats-new-in-gutenberg-26th-june/) describing the changes and performing a [performance audit](./testing-overview.md#performance-testing) is published.

If critical bugs are discovered on stable versions of the plugin, patch versions can be released at any time.

### Release Tool

> > Note that at the time of writing, the tool doesn't support releasing consecutive RC releases. However, it is possible to use the tool for patch releases following the first stable release.

Reviewer 3: Using a blockquote here doesn't seem right to me.

Reviewer 2: This is my thinking: we're using a similar approach in [other parts of the handbook](https://developer.wordpress.org/block-editor/developers/themes/block-based-themes/) to show notices (so they don't interrupt the main flow of the section but are visible). Does that make more sense? An alternative approach I'd be ok with is to add this note as a paragraph by the end of the section.

Reviewer 4: I think the blockquote is fine, it is used often for this case across our docs.
It is too bad we don't have gutenberg blocks in markdown ;-)

Reviewer 3: From an accessibility perspective, it's wrong to use blockquotes for things that aren't actually quotes. So if we want our docs to follow a11y guidelines, we should change all those instances of misused blockquotes at some point.
